tests/v2/test_waku_enr.nim

```diff
@@ -251,3 +252,172 @@ suite "Waku ENR - Multiaddresses":
       multiaddrs.contains(expectedAddr1)
       multiaddrs.contains(addr2)
 
+
+suite "Waku ENR - Relay static sharding":
+
+  test "new relay shards field with single invalid index":
+    ## Given
+    let
+      shardCluster: uint16 = 22
+      shardIndex: uint16 = 1024
+
+    ## When
+    expect Defect:
+      discard RelayShards.init(shardCluster, shardIndex)
+
+  test "new relay shards field with single invalid index in list":
+    ## Given
+    let
+      shardCluster: uint16 = 22
+      shardIndices: seq[uint16] = @[1u16, 1u16, 2u16, 3u16, 5u16, 8u16, 1024u16]
+
+    ## When
+    expect Defect:
+      discard RelayShards.init(shardCluster, shardIndices)
+
+  test "new relay shards field with single valid index":
+    ## Given
+    let
+      shardCluster: uint16 = 22
+      shardIndex: uint16 = 1
+
+    let topic = NsPubsubTopic.staticSharding(shardCluster, shardIndex)
+
+    ## When
+    let shards = RelayShards.init(shardCluster, shardIndex)
+
+    ## Then
+    check:
+      shards.cluster == shardCluster
+      shards.indices == @[1u16]
+
+    let topics = shards.topics.mapIt($it)
+    check:
+      topics == @[$topic]
+
+    check:
+      shards.contains(shardCluster, shardIndex)
+      not shards.contains(shardCluster, 33u16)
+      not shards.contains(20u16, 33u16)
+
+      shards.contains(topic)
+      shards.contains("/waku/2/rs/22/1")
+
+  test "new relay shards field with repeated but valid indices":
+    ## Given
+    let
+      shardCluster: uint16 = 22
+      shardIndices: seq[uint16] = @[1u16, 2u16, 2u16, 3u16, 3u16, 3u16]
+
+    ## When
+    let shards = RelayShards.init(shardCluster, shardIndices)
+
+    ## Then
+    check:
+      shards.cluster == shardCluster
+      shards.indices == @[1u16, 2u16, 3u16]
+
+  test "cannot decode relay shards from record if not present":
+    ## Given
+    let
+      enrSeqNum = 1u64
+      enrPrivKey = generatesecp256k1key()
+
+    let record = EnrBuilder.init(enrPrivKey, enrSeqNum).build().tryGet()
+
+    ## When
+    let typedRecord = record.toTyped()
+    require typedRecord.isOk()
+
+    let fieldOpt = typedRecord.value.relaySharding
+
+    ## Then
+    check fieldOpt.isNone()
+
+  test "encode and decode record with relay shards field (EnrBuilder ext - indices list)":
+    ## Given
+    let
+      enrSeqNum = 1u64
+      enrPrivKey = generatesecp256k1key()
+
+    let
+      shardCluster: uint16 = 22
+      shardIndices: seq[uint16] = @[1u16, 1u16, 2u16, 3u16, 5u16, 8u16]
+
+    let shards = RelayShards.init(shardCluster, shardIndices)
+
+    ## When
```

Comment:
> perhaps a nitpick but `withSharding` instead `withWakuRelaySharding`?
> so `.withSharding().withCapabilities().withMultiAddress()` etc.

If you add this chained builder pattern, you lose the possibility of one of those methods to return a `Result`, which is very convenient.

Reply:
> so we avoid having to init `RelayShards` first? and then passing it?

You can always extend the implementation once this part is used. I added the methods I will need as part of the static sharding discv5 implementation. I am reluctant to add more convenience methods just because they "might be handy".